pkg/config/config.go

```diff
@@ -141,6 +142,21 @@ func New(opts ...Option) (*Config, error) {
 		}
 	}
 
+	// ensure the users hasn't provided conflicting flag and config values
+	// e.g. bacalhau serve --config=api.host=0.0.0.0 --api-host=0.0.0.0 should be rejected.
+	for name, flags := range c.flags {
+		for _, flag := range flags {
+			for key, value := range c.values {
+				if flag.Changed {
+					if name == key {
+						return nil, bacerrors.New("flag: --%s and config flag key %q cannot both be provided. Only one may be used", flag.Name, key).
+							WithHint("Remove --%s or --confg/-c %s=%s from the command", flag.Name, key, value)
+					}
+				}
+			}
```

Comment:
a bit confusing and difficult to read through. Can use better var names. Also I am sure we don't need three for loops here

Reply:
Also please add unit tests